bitbots_vision/src/bitbots_vision/vision.py

```diff
@@ -302,11 +302,9 @@ def handle_image(self, image_msg):
             # self.debug_image_dings.draw_points(
             #     self.line_detector.get_nonlinepoints(),
             #     (0, 0, 0))
-            # debug_image_dings.draw_line_segments(line_detector.get_linesegments(), (180, 105, 255))
-            if self.debug_image:
-                self.debug_image_dings.imshow()
-            if self.debug_image_msg:
-                self.pub_debug_image.publish(self.bridge.cv2_to_imgmsg(self.debug_image_dings.get_image(), 'bgr8'))
+
+            # publish debug image
+            self.pub_debug_image.publish(self.bridge.cv2_to_imgmsg(self.debug_image_dings.get_image(), 'bgr8'))
```

Comment:
You still need an "if" here.

Reply:
No, because the `if`-clause in line 258  decides whether a debug image should be drawn and also published.